### ISSUE TITLE
[BUG] Firewall Daemon Error on Management Node Configuration

### ISSUE THREAD

**Issue by allancarlosvr** [bluebanquise/bluebanquise#998]
This issue highlights a failure in the bluebanquise.infrastructure.firewall role when configuring the firewall on the management node in a lab setup. The management node is the gateway for the compute node, making proper firewall configuration essential for the environment's functionality and security. This bug disrupts the playbook execution, preventing successful role application.

## Context

The lab environment consists of:

1. A management node (management2) with the IP 10.76.0.1, serving as the gateway.
2. A compute node (c001) with the IP 10.76.0.10 on the internal network (net-admin).

The objective is to configure the management node's firewall using the BlueBanquise firewall role to:

- Enable routing between the compute node and external networks.
- Provide security for the lab environment.

The playbook fails due to the firewalld service not running, even though it is a prerequisite for immediate actions by the role.

## Process

1. Create an Ansible inventory file representing the lab setup.
2. Define variables for the firewall configuration in the `group_vars/all` directory **(e.g., firewall.yml).**
3. Use the `bluebanquise.infrastructure.firewall` role in a playbook applied to the management2 host.
4. Run the playbook and observe the error during execution.

## Expected result

The `bluebanquise.infrastructure.firewall` role successfully configures the firewall on `management2`, enabling the node to act as a gateway for the compute node (`c001`), with traffic flowing securely between the internal network and external resources.

## Current result

The playbook fails with the following error:

```bash
failed: [management2] (item=eth1 -> internal) => changed=false 
  ansible_loop_var: item
  invocation:
    module_args:
      state: enabled
      zone: internal
  item:
    interface: eth1
    ip4: 10.76.0.10
  msg: firewall is not currently running, unable to perform immediate actions without a running firewall daemon
```

## Possible Fix

- Ensure that the firewalld service is enabled and running before executing the playbook. This could be added as a pre-task in the playbook:

```yaml
- name: Ensure firewalld is running
  ansible.builtin.service:
    name: firewalld
    state: started
    enabled: true
```

- Modify the `bluebanquise.infrastructure.firewall` role to check and start the `firewalld` service if it is not running.

## Additional data

1. **Lab Schema**

```
                                    +---------------------+
                                    |     Internet        |
                                    +---------------------+
                                              |
                                     +-------------------+
                                     | Management Node   | 
                                     | (management2)     |
                                     | IP: 10.76.0.1     |
                                     +-------------------+
                                              |
                                     +-------------------+
                                     | Compute Node      |
                                     | (c001)            |
                                     | IP: 10.76.0.10    |
                                     +-------------------+
```

![chart_bluebanquise](https://github.com/user-attachments/assets/e62ef9d2-5004-4423-b4e2-c5bcfc841959)

2. Inventory File Example:

```yaml
all:
  children:
    management:
      hosts:
        management2:
          ansible_host: 10.76.0.1
    compute:
      hosts:
        c001:
          ansible_host: 10.76.0.10
```

3. Playbook Example

```yaml
---
- name: Configure Firewall on Management Node
  hosts: management2
  become: true

  pre_tasks:
    - name: Ensure firewalld is running
      ansible.builtin.service:
        name: firewalld
        state: started
        enabled: true

  roles:
    - role: bluebanquise.infrastructure.firewall
      tags:
        - firewall
  ```

**System Details:**
- Management Node OS: Rocky Linux 9.5 (RHEL9)
- Compute Node OS: Rocky Linux 9.5 (RHEL9)
- Internal Network: `net-admin` (10.76.0.0/24)
- Gateway IP: `10.76.0.1``

**Comment by allancarlosvr** [comment]
I  solved my issue with internet forward  from management2 to c001

1. I deleted my `group_vars/all/firewall.yml`

2. Update `group_vars/all/networks.yml`
Define the firewall zones for your net-admin network in `group_vars\all\networks.yml`. Add the necessary details for firewall.zone.

```yaml
networks:
  net-admin:
    subnet: 10.76.0.0        # Cluster's private subnet
    prefix: 24               # Subnet size (/24 for 255.255.255.0)
    services_ip: 10.76.0.10  # Services IP for BlueBanquise
    dns_server: true
    dhcp_server: true
    firewall:
      zone: internal         # Define the firewall zone for this network
      masquerade: true        # Enable masquerading for this network
```
3. Update inventory/cluster/nodes/management.yml
Specify the os_firewall parameter and ensure the network_interfaces include the correct mapping for net-admin on eth1.

```yaml
all:
  hosts:
    management2:
      network_interfaces:
        - interface: eth1
          ip4: 10.76.0.10
          mac: 08:00:27:79:94:00
          network: net-admin
      os_firewall: true
```
4. Validate and Apply Configuration
Run the playbook to validate and apply the configuration:

```bash
ansible-playbook playbooks/managements.yml -i inventory/ -b --limit management2 --extra-vars "time_profile=server"
$ sudo firewall-cmd --get-active-zones
external
  interfaces: eth0
internal
  sources: 10.76.0.0/24
public
  interfaces: eth1
```

5. Check zone configuration
```bash
$ sudo firewall-cmd --zone=internal --list-all
internal (active)
  target: default
  icmp-block-inversion: no
  interfaces: 
  sources: 10.76.0.0/24
  services: cockpit dhcpv6-client mdns samba-client ssh
  ports: 53/udp 53/tcp 123/udp
  protocols: 
  forward: yes
  masquerade: yes
  forward-ports: 
  source-ports: 
  icmp-blocks: 
  rich rules: 
  ```

**Comment by allancarlosvr** [comment]
I  solved my issue with internet forward  from management2 to c001

1. I deleted my `group_vars/all/firewall.yml`

2. Update `group_vars/all/networks.yml`
Define the firewall zones for your net-admin network in `group_vars\all\networks.yml`. Add the necessary details for firewall.zone.

```yaml
networks:
  net-admin:
    subnet: 10.76.0.0        # Cluster's private subnet
    prefix: 24               # Subnet size (/24 for 255.255.255.0)
    services_ip: 10.76.0.10  # Services IP for BlueBanquise
    dns_server: true
    dhcp_server: true
    firewall:
      zone: internal         # Define the firewall zone for this network
      masquerade: true        # Enable masquerading for this network
```
3. Update inventory/cluster/nodes/management.yml
Specify the os_firewall parameter and ensure the network_interfaces include the correct mapping for net-admin on eth1.

```yaml
all:
  hosts:
    management2:
      network_interfaces:
        - interface: eth1
          ip4: 10.76.0.10
          mac: 08:00:27:79:94:00
          network: net-admin
      os_firewall: true
```
4. Validate and Apply Configuration
Run the playbook to validate and apply the configuration:

```bash
ansible-playbook playbooks/managements.yml -i inventory/ -b --limit management2 --extra-vars "time_profile=server"
$ sudo firewall-cmd --get-active-zones
external
  interfaces: eth0
internal
  sources: 10.76.0.0/24
public
  interfaces: eth1
```

5. Check zone configuration
```bash
$ sudo firewall-cmd --zone=internal --list-all
internal (active)
  target: default
  icmp-block-inversion: no
  interfaces: 
  sources: 10.76.0.0/24
  services: cockpit dhcpv6-client mdns samba-client ssh
  ports: 53/udp 53/tcp 123/udp
  protocols: 
  forward: yes
  masquerade: yes
  forward-ports: 
  source-ports: 
  icmp-blocks: 
  rich rules: 
  ```

**Comment by oxedions** [comment]
Dear @allancarlosvr 

Thank you for this issue.
Also, thank you very much for the details you provided, it helps a lot in understanding your needs. :+1: 

I am surprised by the solution. While indeed os_firewall will trigger the launch of the service, defining masquerading in networks should not have any impact.
Would you have enable masquearding manually ?
In theory, to enable masquerade on the host, you would need to use the firewall_zones variable (https://github.com/bluebanquise/bluebanquise/tree/master/collections/infrastructure/roles/firewall#inventory-configuration)

Also, please be extra careful with the usage of os_firewall variable. Please read this: https://bluebanquise.com/documentation/vocabulary.html#equipment-profiles

Basically, these os_ and hw_ variables should be defined for groups and not hosts directly, expect if your cluster is very small.

I must apology for the current documentation state. It was much more detailed with version 1.X of BlueBanquise, but with recent move to 2 and 3, I wasnt able to produce the new documentation. I am working actively on it for the release of 3.2 :blush:
